relay-server/src/services/store.rs

```diff
@@ -803,19 +812,17 @@ impl StoreService {
         retention: u16,
     ) -> Result<(), StoreError> {
         let payload = item.payload();
-        let encodings = self.global_config.current().options.metric_bucket_encodings;
 
-        for bucket in serde_json::from_slice::<Vec<Bucket>>(&payload).unwrap_or_default() {
-            let view = BucketView::new(&bucket);
+        let global_config = self.global_config.current();
+        let encoder = BucketEncoder::new(&global_config);
 
-            let namespace = MetricResourceIdentifier::parse(view.name())
-                .map(|mri| mri.namespace)
-                .unwrap_or(MetricNamespace::Unsupported);
-            let encoding = encodings.for_namespace(namespace);
+        for mut bucket in serde_json::from_slice::<Vec<Bucket>>(&payload).unwrap_or_default() {
+            let namespace = encoder.prepare(&mut bucket);
 
-            let message =
-                self.create_metric_message(org_id, project_id, encoding, &view, retention);
-            self.send_metric_message(namespace, message)?;
+            let view = BucketView::new(&bucket);
+            let message = self
+                .create_metric_message(org_id, project_id, &encoder, namespace, &view, retention);
```

Comment:
Should we be conservative here and fall back to legacy or array encoding when encoding fails?

Reply:
We could, do you see any scenario where this would make sense? 

I am pretty certain that it can't error in practice since the only error should be coming from the `Write` and we're writing to an in memory buffer, which should never fail.